### PR TITLE
docs: Update structured metadata docs

### PR DESCRIPTION
**Pull request by JStickler**
Follow up to the weekly team meeting - Adding some clarification around the use case for using structured metadata.

### CODE REVIEW

File: docs/sources/get-started/labels/structured-metadata.md
Comment (stevendungan) [inline docs/sources/get-started/labels/structured-metadata.md:1]: I think we should replace `trace_id` references with something more representative either infra-specific or an Otel resource attribute

**Review by stevendungan — COMMENTED** [review]
I think we should replace trace_id references with something more representative either infra-specific or an Otel resource attribute

File: docs/sources/get-started/labels/structured-metadata.md
Comment (JStickler) [reply docs/sources/get-started/labels/structured-metadata.md:1]: Looking at the list of "blessed" attributes from #12180 and comparing it to the list of attributes discussed in the design doc, maybe `container.id` or `browser.platform`?

Comment (stevendungan) [reply docs/sources/get-started/labels/structured-metadata.md:1]: I'd propose `container_id` (replace the `.` with `_`), @slim-bean or @sandeepsukhani speak if you have any objection

**Review by slim-bean — APPROVED** [review]

**Pull request merged**